### PR TITLE
Fix the rendering of basis blades in the printer

### PR DESCRIPTION
**Pull request by eric-wieser**
This introduces new sympy objects to represent the basis blades, so that we can define printers for these specifically.

It also implements the sympy "pretty" printer.

This fixes all the issues in #263 except the one requiring sympy/sympy#17926:

image

Still needs:
* [x] Docstrings
* [x] Gutting of the bits of printer intended to support this
* [x] Support for sympy 1.3
* [ ] Possibly better names?

### CODE REVIEW

**Comment by eric-wieser** [comment]
Opened https://github.com/github/codeql/issues/3450 to track the LGTM failure, will delete the comments from now on unless they contain something different.

File: examples/ipython/printing-galgebra.ipynb [hunk 1]
```diff
@@ -129,6 +136,119 @@
...
+     "data": {
+      "text/latex": [
+       "$\\displaystyle \\left( \\left[ \\mathbf{e_{1}}, \\  \\mathbf{e_{2}}, \\  \\mathbf{e_{3}}\\right], \\  \\left[\\begin{matrix}\\left (\\mathbf{e_{1}}\\cdot \\mathbf{e_{1}}\\right )  & \\left (\\mathbf{e_{1}}\\cdot \\mathbf{e_{2}}\\right )  & \\left (\\mathbf{e_{1}}\\cdot \\mathbf{e_{3}}\\right ) \\\\\\left (\\mathbf{e_{1}}\\cdot \\mathbf{e_{2}}\\right )  & \\left (\\mathbf{e_{2}}\\cdot \\mathbf{e_{2}}\\right )  & \\left (\\mathbf{e_{2}}\\cdot \\mathbf{e_{3}}\\right ) \\\\\\left (\\mathbf{e_{1}}\\cdot \\mathbf{e_{3}}\\right )  & \\left (\\mathbf{e_{2}}\\cdot \\mathbf{e_{3}}\\right )  & \\left (\\mathbf{e_{3}}\\cdot \\mathbf{e_{3}}\\right ) \\end{matrix}\\right]\\right)$"
+      ],
+      "text/plain": [
+       "⎛              ⎡(e₁⋅e₁)  (e₁⋅e₂)  (e₁⋅e₃)⎤⎞\n",
+       "⎜              ⎢                         ⎥⎟\n",
+       "⎜[e₁, e₂, e₃], ⎢(e₁⋅e₂)  (e₂⋅e₂)  (e₂⋅e₃)⎥⎟\n",
+       "⎜              ⎢                         ⎥⎟\n",
+       "⎝              ⎣(e₁⋅e₃)  (e₂⋅e₃)  (e₃⋅e₃)⎦⎠"
```
Comment (eric-wieser) [inline examples/ipython/printing-galgebra.ipynb:215]: Subscripts now appear in the pretty output, even in the dot product "symbols"

File: examples/ipython/LaTeX.ipynb [hunk 2]
```diff
@@ -1535,21 +1535,20 @@
...
...t ( e_{u}\\cdot e_{u}\\right ) ^{2} \\left ( e_{v}\\cdot e_{v}\\right ) ^{2} - 2 \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{u}\\cdot e_{v}\\right ) ^{2} \\left ( e_{v}\\cdot e_{v}\\right )  + \\left ( e_{u}\\cdot e_{v}\\right ) ^{4}} \\boldsymbol{e}_{u} + \\frac{\\left ( e_{u}\\cdot e_{u}\\right ) ^{2} A_{vv} B_{vv} - \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{u}\\cdot e_{v}\\right )  A_{vu} B_{vv} - \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{u}\\cdot e_{v}\\right )  A_{vv} B_{uv} - \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{u}\\cdot e_{v}\\right )  A_{vv} B_{vu} + \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{v}\\cdot e_{v}\\right )  A_{vu} B_{uv} + \\left ( e_{u}\\cdot e_{v}\\right ) ^{2} A_{vu} B_{vu} + \\left ( e_{u}\\cdot e_{v}\\right ) ^{2} A_{vv} B_{uu} - \\left ( e_{u}\\cdot e_{v}\\right )  \\left ( e_{v}\\cdot e_{v}\\right )  A_{vu} B_{uu}}{\\left ( e_{u}\\cdot e_{u}\\right ) ^{2} \\left ( e_{v}\\cdot e_{v}\\right ) ^{2} - 2 \\left ( e_{u}\\cdot e_{u}\\right )  \\left ( e_{u}\\cdot e_{v}\\right ) ^{2} \\left ( e_{v}\\cdot e_{v}\\right )  + \\left ( e_{u}\\cdot e_{v}\\right ) ^{4}} \\boldsymbol{e}_{v}  \\end{array} \\right \\} \\end{equation*}\n",
```
Comment (eric-wieser) [inline examples/ipython/LaTeX.ipynb:1551]: This one was removed - sympy 1.3 decided to do some substantial `-` rearrangment compared to 1.4 and 1.5. Fixing that with a regex is totally non-viable for a 64kB line.

File: galgebra/atoms.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+class BasisVectorSymbol(Symbol):
+    """ A symbol representing a basis vector """
+    is_commutative = False
+
+    def _sympystr(self, print_obj):
+        return printer.Eprint.Base(print_obj._print_Symbol(self))
+
+    def _latex(self, print_obj):
+        try:
+            return print_obj._print_Symbol(self, style="bold")
```
Comment (eric-wieser) [inline galgebra/atoms.py:34]: @asmeurer: Is this public API, or should we be bolding this ourselves?

**Comment by utensil** [comment]
It’s a bit weird with the commit log: There’s only one commit and its commit log feels like only one of many commits.

**Comment by eric-wieser** [comment]
What do you mean by that?

**Comment by eric-wieser** [comment]
Oh, now I see what you mean. Will fix

**Comment by eric-wieser** [comment]
Fixed, there are now two commits.

Failure on sympy master is because of my upstream change to sympy, will attempt to nbsanitize it

**Pull request merged**